### PR TITLE
1x4p1

### PR DESCRIPTION
**Pull request by daviddoan**
## Description

Added a new macropad ([1x4]+1 Macropad), designed by Array Peripherals.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/arrayperipherals/1x4p1/1x4p1.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "1x4p1.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/arrayperipherals/1x4p1/1x4p1.c:22]:
```suggestion
```

File: keyboards/arrayperipherals/1x4p1/1x4p1.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#include "quantum.h"
+
+#define LAYOUT_ortho_1x5( \
+    k01, k02, k03, k04, k05\
+  ) { \
+    { k01, k02, k03, k04, k05} \
+}
+
+#define LAYOUT LAYOUT_ortho_1x5
```
Comment (drashna) [inline keyboards/arrayperipherals/1x4p1/1x4p1.h:28]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/arrayperipherals/1x4p1/config.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+/* key matrix size */
+#define MATRIX_ROWS 1
+#define MATRIX_COLS 5
+
+
+#define DIRECT_PINS {   \
+    { C7, B7, D6, F5, F7} \
+}
+//speed for double tap
+#define TAPPING_TERM 200
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/config.h:38]:
```suggestion
```

File: keyboards/arrayperipherals/1x4p1/config.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#define UNUSED_PINS
+
+/* rotary encoder*/
+#define ENCODERS_PAD_A {F0}
+#define ENCODERS_PAD_B {F1}
+#define ENCODER_RESOLUTION 4 //default/suggested
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/config.h:44]:
```suggestion
```

File: keyboards/arrayperipherals/1x4p1/info.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "keyboard_name": "[1 x 4] + 1 Macropad",
+  "url": "https://github.com/daviddoan",
+  "maintainer": "David Doan",
+  "width": 5,
+  "height": 1,
+  "layouts": {
+    "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/info.json:8]:
```suggestion
    "LAYOUT_ortho_1x5": {
```

File: keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+
+//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { //button closest to usb is first
+  [_MAIN] = LAYOUT(
+     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
+   ),
+
+  [1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c:38]:
```suggestion
  [_MAIN] = LAYOUT_ortho_1x5(
     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
   ),

  [1] = LAYOUT_ortho_1x5(
```

File: keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+
+//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { //button closest to usb is first
+  [_MAIN] = LAYOUT(
+     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
+   ),
+
+  [1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c:38]:
```suggestion
  [_MAIN] = LAYOUT_ortho_1x5(
     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
   ),

  [1] = LAYOUT_ortho_1x5(
```

File: keyboards/arrayperipherals/1x4p1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+TAP_DANCE_ENABLE = no      # Double press does something different
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/arrayperipherals/1x4p1/rules.mk:25]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
UNICODE_ENABLE = yes
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/arrayperipherals/1x4p1/config.h [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    { C7, B7, D6, F5, F7} \
+}
+#define UNUSED_PINS
+
+/* rotary encoder*/
+#define ENCODERS_PAD_A {F0}
+#define ENCODERS_PAD_B {F1}
+
+//This is for VIA configurator
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/config.h:44]:
```suggestion
```
I don't see a reason to change this from the default (4) and even then, I'd recommend putting such defines in the `via` keymap's `config.h`.

File: keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#define _MAIN 0
+
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c:21]:
```suggestion
```
I recommend either using an `enum` to name all layers or name none of them.

File: keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      tap_code(KC_MS_WH_UP);
+    } else {
+      tap_code(KC_MS_WH_DOWN);
+    }
+  }
+}
+
+//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { //button closest to usb is first
+  [_MAIN] = LAYOUT_ortho_1x5(
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c:34]:
```suggestion
  [0] = LAYOUT_ortho_1x5(
```

File: keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
+   ),
+
+  [1] = LAYOUT_ortho_1x5(
+  	 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(1)
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/default/keymap.c:39]:
```suggestion
  	 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(0)
```

File: keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#define _MAIN 0
+
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c:21]:
```suggestion
```

File: keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+      tap_code(KC_MS_WH_UP);
+    } else {
+      tap_code(KC_MS_WH_DOWN);
+    }
+  }
+}
+
+//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { //button closest to usb is first
+  [_MAIN] = LAYOUT_ortho_1x5(
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c:34]:
```suggestion
  [0] = LAYOUT_ortho_1x5(
```

File: keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+     KC_ESC, KC_TAB, KC_LSHIFT, KC_LCTRL, TG(1)
+   ),
+
+  [1] = LAYOUT_ortho_1x5(
+     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(1)
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c:39]:
```suggestion
     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(2)
```

File: keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+  )
```
Comment (Erovia) [inline keyboards/arrayperipherals/1x4p1/keymaps/via/keymap.c:40]:
```suggestion
  ),
  [2] = LAYOUT_ortho_1x5(
     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(3)
  ),
  [3] = LAYOUT_ortho_1x5(
     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(0)
  )
```

**Review by Erovia — CHANGES_REQUESTED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**